src/basics/custom.md

```diff
@@ -0,0 +1,493 @@
+# Custom messages
+
+Blockchain creators might define chain-specific logic triggered through defined by them messages.
+`CosmWasm` provides a way to send such messages through `cosmwasm_std::CosmosMsg::Custom(..)` variant.
+
+`Sylvia` supports this feature, allowing both custom interfaces and contracts.
+
+## Custom interface 
+
+To make further code examples simpler to read, let's consider we have the custom query and exec defined as such:
+
+```rust
+pub mod external_messages {
+    use cosmwasm_schema::cw_serde;
+    use cosmwasm_std::{CustomMsg, CustomQuery};
+
+    #[cw_serde]
+    pub enum ExternalMsg {
+        Poke,
+    }
+
+    #[cw_serde]
+    pub enum ExternalQuery {
+        IsPoked,
+    }
+
+    impl CustomQuery for ExternalQuery {}
+    impl CustomMsg for ExternalMsg {}
+}
+```
+
+Notice that query has to implement `cosmwasm_std::CustomQuery`, and the message has to implement `cosmwasm_std::CustomMsg`.
+
+Now that we have our messages defined, we should consider if our new interface is meant to work only on a specific chain
+or if the developer implementing it should be free to choose which chain it should support.
+
+To enforce support for the specific chain, the user has to use `#[sv::custom(msg=.., query=..)]` attribute.
+Once `msg` or `query` is defined, it will be enforced that all messages or queries used in the interface
+will use them. This is necessary for `dispatch` to work.
+
+```rust
+use crate::external_messages::{ExternalMsg, ExternalQuery};
+use cosmwasm_std::{Response, StdError};
+use sylvia::interface;
+use sylvia::types::{ExecCtx, QueryCtx};
+
+#[interface]
+#[sv::custom(msg=ExternalMsg, query=ExternalQuery)]
+pub trait SvCustom {
+    type Error: From<StdError>;
+
+    #[msg(exec)]
+    fn sv_custom_exec(
+        &self,
+        ctx: ExecCtx<ExternalQuery>,
+    ) -> Result<Response<ExternalMsg>, Self::Error>;
+
+    #[msg(query)]
+    fn sv_custom_query(&self, ctx: QueryCtx<ExternalQuery>) -> Result<String, Self::Error>;
+}
+```
+
+If, however, we would like to give the developers the freedom to choose which chain to support, we can use define
+the interface with the associated type instead.
+
+```rust
```

Comment:
I would add here imports explicitly:
```rust
use cosmwasm_std::{CustomMsg, CustomQuery, Response, StdError};
use sylvia::interface;
use sylvia::types::{ExecCtx, QueryCtx};
```
because they differ from the example with `SvCustom` trait.